src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,631 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/TrainingOnlyEstimatorImpl.h"
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wold-style-cast"
+#   pragma clang diagnostic ignored "-Wsign-conversion"
+#   pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#   pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#   pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#   pragma clang diagnostic ignored "-Wmissing-noreturn"
+#   pragma clang diagnostic ignored "-Wdocumentation"
+#   pragma clang diagnostic ignored "-Wdouble-promotion"
+#   pragma clang diagnostic ignored "-Wcast-align"
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#   pragma clang diagnostic ignored "-Wshadow"
+#elif (defined _MSC_VER)
+#   pragma warning(push)
+#   pragma warning(disable: 4127)
+#endif
+
+#include "../3rdParty/eigen/Eigen/Dense"
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#elif (defined _MSC_VER)
+#   pragma warning(pop)
+#endif
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+static constexpr char const * const         SVDComponentsEstimatorName("SVDComponentsEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         SVDComponentsAnnotationData
+///  \brief         Contains SVD components: Eigenvalues and Eigenvectors
+template <typename T>
+class SVDComponentsAnnotationData {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+
+    T const                                          SingularValues;
+    T const                                          SingularVectors;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SVDComponentsAnnotationData(T singularvalues, T singularvectors);
+    ~SVDComponentsAnnotationData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(SVDComponentsAnnotationData);
+};
+
+namespace ComponentsDetails {
+
+namespace {
```

Comment:
This also has a meaning of static, unique to the current unit of compilation. You have already introduced ComponentDetails namespace for the purpose. What is the reason for nameless namespaces? What will it do in difference CC files where it is included?